packages/sdk/src/interaction/interaction.js

```diff
@@ -136,46 +142,45 @@ export const Bad = (ix, reason) => {
   return ix
 }
 
-const makeIx = (wat) => (ix) => {
+const makeIx = wat => ix => {
   ix.tag = wat
   return Ok(ix)
 }
 
-const makeAccount = (acct, tempId) => (ix) => {
-  ix.accounts[tempId] = JSON.parse(ACCT)
-  ix.accounts[tempId].tempId = tempId
-  ix.accounts[tempId].addr = acct.addr
-  ix.accounts[tempId].keyId = acct.keyId
-  ix.accounts[tempId].sequenceNum = acct.sequenceNum
-  ix.accounts[tempId].signature = acct.signature
-  ix.accounts[tempId].signingFunction = acct.signingFunction
-  ix.accounts[tempId].resolve = acct.resolve
-  ix.accounts[tempId].role = {
-    ...ix.accounts[tempId].role,
-    ...acct.role,
+export const prepAccount = (acct, opts = {}) => ix => {
+  invariant(
+    typeof acct === "function" || typeof acct === "object",
+    "prepAccount must be passed an authorization function or an account object"
+  )
+  invariant(opts.role != null, "Account must have a role")
+
+  const ACCOUNT = JSON.parse(ACCT)
+  const role = opts.role
+  const tempId = uuid()
+
+  acct = typeof acct === "function" ? {resolve: acct} : acct
+
+  ix.accounts[tempId] = {
+    ...ACCOUNT,
+    tempId,
+    ...acct,
+    role: {
+      ...ACCOUNT.role,
+      ...(typeof role === "object" ? role : {}),
```

Comment:
Can remove this check I think, if approved by @orodio and @JeffreyDoyle. 
Wasn't totally clear on the direction of role vs. roles or if it will ever be an object.